### ISSUE TITLE
Unresponsive/unrecognized controls in Retroarch Wii

### ISSUE THREAD

**Issue by Cee123** [libretro/RetroArch#11217]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

With this latest version (or versions?) the controls for the Wii are unrecognized by a bunch of cores, controls are unmappable and won't respond during game. I'm not sure if it's just mine, or the issue is with everybody's.

### Expected behavior

The cores used to load. And the content loads, and then the controllers respond. Some cores still do, but many of them don't.

### Actual behavior

Unresponsive controls, you cannot even map the controls because the buttons for the Wiimote or Nunchuck and possibly even the classic controller won't register. But strangely the retroarch menu still works, and you can still access the menu with the home button, so it's completely baffling!

### Steps to reproduce the bug

I have nightly versions from both July and August, and the cores seem to do it on these versions. I've also compiled a couple of cores myself using a virtual machine running Ubuntu and same thing.

### Bisect Results

I'm not sure it just happened all of a sudden. I've tried deleting the retroarch cfg files and letting them generate again, and tried replacing the boot.dol file with different versions thinking it could be that.

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.9.0

### Environment information

Wii

**Comment by twinaphex** [comment]
We lack a lot of active Wii coders right now.

**Comment by Cee123** [comment]
Yeah that's no problem. Plus the buildbot was hacked which sucks. Just thought I'd mention it here in case i forget. Seems strange. It might just be the setup on mine, but we'll see if others have the same issue.

**Comment by Cee123** [comment]
After some testing I determined which cores they are as of the latest nightly I tried 13/08/2020.

Working fine
- 2048
- Cannonball
- FCEumm
- Gambatte
- Genesis Plus Gx
- Mednafen Lynx
- Mednafen Supergrafx
- Nestopia
- Prboom
- QuickNes
- SMS Plus (built from source)
- Tyrquake

Controls not responding
- Fixgb
- Fixnes
- Handy (built from source)
- Mame2000
- Mednafen pce (built from source)
- Mednafen pce fast
- Mednafen vb
- Mgba
- MrBoom
- NXEngine
- Prosystem (built from source)
- Snes9x
- Snes9x2005
- Snes9x2005_plus
- Snes9x2010
- Stella
- Stella2014
- Vba Next

Crash to main screen
- FB Alpha 2012
- FB Alpha2012 Cps1
- FB Alpha2012 Cps2
- FB Alpha 2012 Neo Geo
- ECWolf

Crash dumping
- FB Alpha2012 Cps3

Black Screen
- Dosbox svn
- Snes9x2002

Not tested
- 81
- Atari800
- Bk
- Cap32
- Crocods
- Freechaf
- Freeintv
- Fuse
- Gme
- Lutro
- Mednafen Ngp
- Mednafen Wswan
- Mu
- Pocketcdg
- Pokemini
- Quasi88
- Squirreljme
- Theodore
- Uzem
- X1

**Comment by Cee123** [comment]
Good news, the issue was identified here as the input bitmask. Seems like it's missing now on most of the cores for Wii (and possibly Wii U).
https://github.com/libretro/libretro-handy/issues/75

**Comment by negativeExponent** [comment]
input bitmasks has never been a defacto input polling method. dunno whats going on here.

**Comment by trioan** [comment]
@Cee123 
```Snes9x2010```

Can you verify this core does not work with Wii controls? It has input bitmask support already. :o

**Comment by Cee123** [comment]
> @Cee123
> 
> Can you verify this core does not work with Wii controls? It has input bitmask support already but is in your non-working list.. :o
> https://github.com/libretro/snes9x2010
> 
> [libretro/snes9x2010@1bbbd3e](https://github.com/libretro/snes9x2010/commit/1bbbd3e253b3a25153bfd911f2baa46a5e6c5bb2)
> [libretro/snes9x2010@0167161](https://github.com/libretro/snes9x2010/commit/01671616de0f5eaaea92aabb4ea33b1a57107949)

Last time i tried it the controller wasnt working which is weird if it does have the input bitmask. I'll test it again because i think i also had a remap file set for that core. I'll delete that and test it again.

**Comment by Cee123** [comment]
> 
> 
> > @Cee123
> > Can you verify this core does not work with Wii controls? It has input bitmask support already but is in your non-working list.. :o
> > https://github.com/libretro/snes9x2010
> > [libretro/snes9x2010@1bbbd3e](https://github.com/libretro/snes9x2010/commit/1bbbd3e253b3a25153bfd911f2baa46a5e6c5bb2)
> > [libretro/snes9x2010@0167161](https://github.com/libretro/snes9x2010/commit/01671616de0f5eaaea92aabb4ea33b1a57107949)
> 
> Last time i tried it the controller wasnt working which is weird if it does have the input bitmask. I'll test it again because i think i also had a remap file set for that core. I'll delete that and test it again.

I grabbed the snes9x2010 from the latest nightly and the controls do work now. Which is strange because they wouldn't work in an earlier release/nightly.

**Comment by trioan** [comment]
Going to pretend that was some compiler corruption for now, until it happens again. Could be related to the crashing issues.

Any priority cores you'd like bitmasks to be added faster? Otherwise I'll just keep randomly pick some cores that are handled by twinaphex first.

**Comment by twinaphex** [comment]
The more cores bitmasks are added to, the better. It will result in faster input handling due to only having to do one API call for grabbing the 16 retro gamepad button states at once.

**Comment by trioan** [comment]
While working on Mr. Boom, had this very strange thing happen with input bitmask. Start button works until I got to the menu. Then it stopped working but the other buttons looked okay. Had to use a ```state = state_joypad & (1 << id) ? 1 : 0;``` and then it started working again. Using raw ```input_state_cb``` worked okay.

So I will have to relook over all the bitmask input cores and force it. And maybe do some more cleanup.

**Comment by trioan** [comment]
@Cee123 
Could you give an update on the list once the buildbot starts pushing some of these updates? Does snes9x2002 still black screen crash though?

**Comment by Cee123** [comment]
> 
> 
> While working on Mr. Boom, had this very strange thing happen with input bitmask. Start button works until I got to the menu. Then it stopped working but the other buttons looked okay. Had to use a `state = state_joypad & (1 << id) ? 1 : 0;` and then it started working again. Using raw `input_state_cb` worked okay.
> 
> So I will have to relook over all the bitmask input cores and force it. And maybe do some more cleanup.

That's really strange. Not sure what would've caused that.

**Comment by Cee123** [comment]
> 
> 
> @Cee123
> Could you give an update on the list once the buildbot starts pushing some of these updates? Does snes9x2002 still black screen crash though?

Yeah sure thing! I'll test them out. They should be working good now. I think it still does, although I'm not sure. I'll test it again.

**Comment by trioan** [comment]
Mr. Boom wants 0 or 1 some places. Other times it doesn't care.

**Comment by Cee123** [comment]
Fantastic work. I just tried a bunch of cores and the controls are all working for the ones I've mentioned in that list above. Except for these couple of cores.
- fixgb
- fixnes
- mame2000
- mednafen_supergrafx (was working before, weird!)

Can also confirm snes9x2002 still doesn't work. Neither does dosbox_svn.

Also none of the fbalpha2012 cores are working anymore. I'm thinking it had a lot to do with when the buildbot got hacked and things got corrupted.

**Comment by trioan** [comment]
NXEngine glitches being input or graphics?


Found the missing core links here.
https://raw.githubusercontent.com/libretro/libretro-super/master/recipes/nintendo/wiiu


I didn't touch SuperGrafx but latest commit log
https://github.com/libretro/beetle-supergrafx-libretro/commits/master

**Comment by trioan** [comment]
@Cee123 
I'm unsure if the FIX94 cores will get merged in but you're welcome to build the patches yourself, with the tagged links above this post.

Glad to see we made some decent headway. I know there's lots more untested cores without bitmask, but not sure I'll be the one to go through them all. The full list would make normal people cry a bit.
https://raw.githubusercontent.com/libretro/libretro-super/master/recipes/windows/cores-windows-x64_seh-generic

I'll give a decent look into the remaining Wii problems as a continued favor, since you actually use these cores. And well, there's few people around to help with your ongoing assistance.

**Comment by metallic77** [comment]
atari800 is also possible to compile and controls are not working, bluemsx too and fmsx

https://github.com/libretro/libretro-atari800/issues/55

**Comment by i30817** [comment]
Does this apply to the dolphin core? I've been trying to work upstream to get the EGL/DRM pr to work on KMS and i'm discouraged how Xinput seems necessary to get both the pointer (eg: touchpad) and buttons (eg: keyboard) working at once there (there is no xinput backend for DRM/KMS, since it depends on X11).

**Comment by metallic77** [comment]
dolphin core wouldn't even dare to start on the Wii.

**Comment by i30817** [comment]
Well i meant the bitmask controls. I'll be honest i asked because i was trying to patch upstream yesterday to get the EGL/DRM pull request there working with hotkeys. I succeeded (sort of, modifier keys don't work). However i found something that derailed the effort: any control backend but xinput2 upstream 'virtual core pointer' doesn't allow the control of the keyboard keys  and the touchpad at the same time (and evdev had a bunch of touble with the touchpad). Since X11 doesn't work in wayland this means that wii games that make use of the wiimote pointer are lost there.
Then i tried the libretro core and realized the exact same thing happens there with the emulated wiimote.

**Comment by trioan** [comment]
Bitmask input only affects reading digital retro_joypad buttons, so I think anything else "analog" will still fail with Wiimote. Although I don't see bitmasks being used.
https://github.com/libretro/dolphin/blob/master/Source/Core/DolphinLibretro/Input.cpp


Working on blueMSX and will follow up with fmsx and Atari800 if success.

**Comment by twinaphex** [comment]
Do we have the FIX94 cores in the libretro organization right now? If not, I can clone them and a PR could be submitted there.

**Comment by trioan** [comment]
I did not see FIX94 cores in libretro organization. When I read the Wii recipe list, then I realized what Cee123 was referring to.

**Comment by metallic77** [comment]
fmsx and atari800 compile and work properly now

**Comment by trioan** [comment]
@Cee123 
If mednafen_supergrafx still fails, go ask negativeExponent who is actively maintaining that core. As for fbalpha, hmm. Otherwise I think mostly everything you use is covered.

**Comment by Cee123** [comment]
> 
> 
> @Cee123
> If mednafen_supergrafx still fails, go ask negativeExponent who is actively maintaining that core. As for fbalpha, hmm. Otherwise I think mostly everything you use is covered and waiting in queue. Forgot about ECWolf.

Awesome, thanks so much! You've been a great help and done good work. All the cores I've tried are working great now, the controls all work and everything. Although it does make me wonder if retroarch for the Wii U would have the same controller issues. But other than that, I've compiled the cores not in the latest nightly, even those are working now with the controls. So that all seems to be fixed.

**Comment by Cee123** [comment]
> 
> 
> fmsx and atari800 compile and work properly now

Nice. Two more cores that work on Wii (and Wii U).

**Comment by metallic77** [comment]
Yes, someone needs to fix some c64 core too. I tried and compiled vice_c64 but got black screen after running on my Wii and system hang. Frodo stops compiling with an error on Vic.cpp

https://github.com/libretro/vice-libretro/issues/280

**Comment by Cee123** [comment]
> 
> 
> Yes, someone needs to fix some c64 core too. I tried and compiled vice_c64 but got black screen after running on my Wii and system hang. Frodo stops compiling with an error on Vic.cpp
> 
> [libretro/vice-libretro#280](https://github.com/libretro/vice-libretro/issues/280)

Interesting. I did wonder if that core would work on the Wii.

**Comment by metallic77** [comment]
If you mean having the power, i run Vice 3.2 PSP the other day with some minor frame skip, and i believe this one is 3.0.

**Comment by Cee123** [comment]
> 
> 
> If you mean having the power, i run Vice 3.2 PSP the other day with some minor frame skip, and i believe this one is 3.0.

Nice. Yeah I think it should run on there no problem.

**Comment by trioan** [comment]
It was a good run but I'm not interested in dealing with other code developers anymore. Hopefully you'll have someone to fix up the last remnants.

I've seen other threads with input complaints for Wii but never Wii U so far.

**Comment by Cee123** [comment]
> It was a good run but I'm not interested in dealing with other code developers anymore. Hopefully you'll have someone to fix up the last remnants.
> 
> I've seen other threads with input complaints for Wii but never Wii U so far.

Sorry to hear that. Not sure who you mean. But yeah seeing the attitudes of some (fortunately most have been nice enough though), I dont blame you personally. Sad that you've gone now. But thank you for everything you've done. It's definitely been appreciated.

**Comment by twinaphex** [comment]
Hi there trioan, if you are still reading this, I very much regret this as you were doing excellent work and I would hate to see you go.

Can you please contact me at libretro@gmail.com? Perhaps there is a way we can resolve this, I would very much hope we can at least.

**Comment by metallic77** [comment]
Yep the job done was great. I already had some of the fixed cores working normally from previous RA versions but not Atari800 or fmsx. Practically added 2 more systems on the Wii. The Wii is important because it's a relatively powerful device to output 240p on a CRT without extra hacks and cables or tweaks, well it's not PS4 but at least can run most of the old 8/16bit really well.

**Comment by Cee123** [comment]
I literally have no idea what happened just now. Lol

**Comment by Cee123** [comment]
> Yep the job done was great. I already had some of the fixed cores working normally from previous RA versions but not Atari800 or fmsx. Practically added 2 more systems on the Wii. The Wii is important because it's a relatively powerful device to output 240p on a CRT without extra hacks and cables or tweaks, well it's not PS4 but at least can run most of the old 8/16bit really well.

Yeah same here. I like the Wii. It's awesome how you can turn it into an all in one emulation machine, media player etc.

**Comment by metallic77** [comment]
Well maybe he has other important things to do, but in any case we thank him very much for his job. If it's some other issue would be nice to have him back. Also i believe he deserved a small donation for his work if he is willing to accept it.

As for the Wii, yeah it is an all in one machine and don't forget it can also run GC games which has an amazing library. Basically needs c64-dosbox-Amiga cores to be complete (and usb keyboard support, don't know if it works?)

**Comment by Cee123** [comment]
> Well maybe he has other important things to do, but in any case we thank him very much for his work. If it's some other issue would be nice to have him back. Also i believe he deserved a small donation for his work if he is willing to accept it.
> 
> As for the Wii, yeah it is an all in one machine and don't forget it can also run GC games which has an amazing library. Basically needs c64-dosbox-Amiga cores to be complete (and usb keyboard support, don't know if it works?)

Yeah that backward compatibility with Gamecube is awesome. So you have both wii and Gamecube library. And you can use Nintendont. Also not64 for N64 and wiisxr for PS1. Now that would be fantastic too. Commodore 64, amiga and dosbox should all work pretty well i think. Dosbox just loads to a black screen. I wonder if there's something extra the core needz with the setup.

**Comment by metallic77** [comment]
I compiled again today vice_x64 and dosbox_svn with same results; black screen and system hangs. About the Amiga, hmm maybe need an older version of UAE. Don't know if the current puae can run full speed, Amiga has some demanding games (the hardware has a ton of tricks and custom chips) that you need a good CPU to run well, like Jim Power for example. But i could be wrong. After all i remember i was running Amiga emulation well on systems with 1ghz processor and 128mb ram

**Comment by twinaphex** [comment]
Which cores remain as of this writing that still require attention?

**Comment by metallic77** [comment]
From what i have tested dosbox_svn gives a black screen and system hang after load core (compiled from source+nightlies).
Also some c64 core (even frodo) would be great. I have to test nightly to give feedback which cores don't have working controls.

**Comment by Cee123** [comment]
> 
> 
> Which cores remain as of this writing that still require attention?

As far as I know, out of the ones I've tested at least, I think it's just the fix94 cores and the supergrafx core (but I might have to test the latest nightly version again). Not sure about the others or the cores. I know the fbalpha cores crash as well for some reason.

**Comment by metallic77** [comment]
Seems the asm part at file vic.cpp (line 1422) of frodo needs a rewrite because after disabling it and doing some other crap like disabling SAM etc it compiled and run on my wii without showing any graphics, showing only the leds bar.
I used libretro-frodo master, not r-type fork.

https://github.com/r-type/frodo-libretro/issues/8

**Comment by Cee123** [comment]
I tried to compile picodrive for the wii because i wanted to see if Sega 32x would work on there and also how megadrive etc run with that. But it just came up with a bunch of errors. I think its because that core wasnt ported to other platforms by the developer.

**Comment by metallic77** [comment]
Yes, frodo for example has that part written maybe 15 years ago, and all ifdef powerpc  probably are for PPC Mac and a specific compiler. The syntax is not valid for g++, asm should be followed by (".......") so that the compiler accepts it, and probably there are more errors there.

**Comment by metallic77** [comment]
managed to compile a frodo core that outputs image, by commenting all _ifdef __POWERPC___ inside Vic.cpp and inserting some commonflags to makefile.libretro to bypass some errors.
Probably an experienced coder could fix it in 1 hour i guess.

https://www.dropbox.com/s/mdtrmyj1vzc7bb5/frodo_libretro_wii.zip?dl=0

**Comment by Edu789** [comment]
The cores fbalpha2012_cps1, cps2, cps3, fbalpha2012_libretto and fbalpha2012_neogeo stopped working as of version 1.8.5

**Comment by rogerioronin** [comment]
Sorry for my bad English, the core "**snes9x2002_libretro_wii.dol**" works well now, the problem is the sound, it is very bad!

Thanks for the work, amazing!

**Comment by Cee123** [comment]
> 
> 
> managed to compile a frodo core that outputs image, by commenting all _ifdef **POWERPC**_ inside Vic.cpp and inserting some commonflags to makefile.libretro to bypass some errors (sigaction etc, probably need the same with psp)
> Probably an experienced coder could fix it in 1 hour i guess.
> 
> P.S. controls not working, buttons recognised as mouse, same problem with previous cores
> 
> https://www.dropbox.com/s/mdtrmyj1vzc7bb5/frodo_libretro_wii.zip?dl=0

You're right. I can imagine both Commodore 64 and Amiga working on the Wii. I've been toying around with both emulators after you've mentioned them, and I've finally managed to get them to work on my PC. They work great with little hardware demand! I can imagine both working on the Wii. If you manage to get them to work on there that would be awesome.

**Comment by Cee123** [comment]
> 
> 
> Sorry for my bad English, the core "**snes9x2002_libretro_wii.dol**" works well now, the problem is the sound, it is very bad!
> 
> Thanks for the work, amazing!

Oh really? Great! I'll have to try it with the latest nightly. Thanks for letting us know. I wonder if Dosbox works now as well.

**Comment by rogerioronin** [comment]
In my tests the core "atari800_libretro_wii.dol", I only tested games of "Atari 5200" and the control didn't work, I'm using an Original Classic Controller!

**Comment by metallic77** [comment]
I tried and compiled hatari (Atari ST) emulator too, but when starting... black screen - system hang, same with vice_x64. 
I believe vice_x64, frodo, hatari could run well on Wii. At the very least, frodo could work with controls fixed.

**Comment by metallic77** [comment]
https://www.dropbox.com/s/bed73d3ti87lxzs/frodo_libretro_wii.zip?dl=0

I did a frodo core with controls working. Check this out

**Comment by Edu789** [comment]
> > Sorry for my bad English, the core "**snes9x2002_libretro_wii.dol**" works well now, the problem is the sound, it is very bad!
> > Thanks for the work, amazing!
> 
> Oh really? Great! I'll have to try it with the latest nightly. Thanks for letting us know. I wonder if Dosbox works now as well.

In the Retroarch info file, it says that Dosbox needs the files libsdl1.2 and libsdl-net1.2. to work.

**Comment by Edu789** [comment]
> I tried to compile picodrive for the wii because i wanted to see if Sega 32x would work on there and also how megadrive etc run with that. But it just came up with a bunch of errors. I think its because that core wasnt ported to other platforms by the developer. It works good on the 3ds though.
If it works on the Playstation 2 retroarch, it should work on the Wii.

**Comment by i30817** [comment]
The wii is big endian and the ps2 is little endian. No it 'shouldn't just work'.

**Comment by Cee123** [comment]
> > > Sorry for my bad English, the core "**snes9x2002_libretro_wii.dol**" works well now, the problem is the sound, it is very bad!
> > > Thanks for the work, amazing!
> > 
> > 
> > Oh really? Great! I'll have to try it with the latest nightly. Thanks for letting us know. I wonder if Dosbox works now as well.
> 
> In the Retroarch info file, it says that Dosbox needs the files libsdl1.2 and libsdl-net1.2. to work.

Oh ok. That must be what im doing wrong.

**Comment by metallic77** [comment]
i made a hatari core for Wii, here

https://gbatemp.net/threads/wii-atari-st-core-hatari-retroarch-wii-new.574656/

**Comment by metallic77** [comment]
Made a Working dosbox, here. Made a pull request too.

https://gbatemp.net/threads/wii-dos-core-dosbox-retroarch-wii-new.574725/

**Comment by Cee123** [comment]
> Made a Working dosbox, here. Made a pull request too.
> 
> https://gbatemp.net/threads/wii-dos-core-dosbox-retroarch-wii-new.574725/

Good work man!

**Comment by rogerioronin** [comment]
In some versions of Retroarch a version of the Neko Project II Kai nucleus was released, but it didn't work, you can't make a correction?

**Comment by Edu789** [comment]
> Yep the job done was great. I already had some of the fixed cores working normally from previous RA versions but not Atari800 or fmsx. Practically added 2 more systems on the Wii. The Wii is important because it's a relatively powerful device to output 240p on a CRT without extra hacks and cables or tweaks, well it's not PS4 but at least can run most of the old 8/16bit really well.


In the options of the atari 800 you have to change the system of the Atari 400/800 to 5200, to enter straight into the game and not on the core screen.
The games work well, but the audio is hissing.

**Comment by Cee123** [comment]
@metallic77 Hey there, any chance you managed to get vice_x64 (Commodore 64) and Puae (Amiga) working on Wii? I reckon both of those should run well on there. I got Vice to compile but I haven't tried it yet. But I couldn't even get Puae to compile for the Wii.

**Comment by Edu789** [comment]
> @metallic77 Hey there, any chance you managed to get vice_x64 and Puae working on Wii? I reckon both of those should run well on there. I got Vice to compile but I couldn't get it to convert to .dol. And I couldn't even get Puae to compile for the Wii.


They added several Commodore cores in the last nightly builds, but when they run they give the black screen ...

**Comment by Cee123** [comment]
> > @metallic77 Hey there, any chance you managed to get vice_x64 and Puae working on Wii? I reckon both of those should run well on there. I got Vice to compile but I couldn't get it to convert to .dol. And I couldn't even get Puae to compile for the Wii.
> 
> They added several Commodore cores in the last nightly builds, but when they run they give the black screen ...

Ah yeah same here. I tried the nice Vice cores for the wii but they just freeze on a black screen when loading the core.

I managed to get DosBox working. I compiled a core and it works.

**Comment by metallic77** [comment]
Dosbox works, they merged the pull requests i did.

**Comment by Cee123** [comment]
> Dosbox works, they merged the pull requests i did.

Oh that's fantastic. Good work! It works well!

**Comment by Cee123** [comment]
Just a couple of other working cores for Wii, not sure why they aren't included.
[prosystem_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776150/prosystem_libretro_wii.zip)
[reminiscence_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776152/reminiscence_libretro_wii.zip)
[smsplus_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776154/smsplus_libretro_wii.zip)
[xrick_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776155/xrick_libretro_wii.zip)

And of course DosBox is all working now.
[dosbox_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776158/dosbox_libretro_wii.zip)

**Comment by Cee123** [comment]
The cores for CPS3 and FixGb and FixNes are not in the nightly anymore. But I still have these old versions.
[fbalpha2012_cps3_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776180/fbalpha2012_cps3_libretro_wii.zip)
[fixgb_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776181/fixgb_libretro_wii.zip)
[fixnes_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776182/fixnes_libretro_wii.zip)

**Comment by IAmBoring** [comment]
FixGB is broken but I patched it. I imagine FixNes also and maybe missing others.

**Comment by Cee123** [comment]
> FixGB is broken but I patched it. I imagine FixNes also and maybe missing others.

Oh ok, that's why they werent included. Nice. They should work now hopefully.

**Comment by rogerioronin** [comment]
> Just a couple of other working cores for Wii, not sure why they aren't included.
> [prosystem_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776150/prosystem_libretro_wii.zip)
> [reminiscence_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776152/reminiscence_libretro_wii.zip)
> [smsplus_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776154/smsplus_libretro_wii.zip)
> [xrick_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776155/xrick_libretro_wii.zip)
> 
> And of course DosBox is all working now.
> [dosbox_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776158/dosbox_libretro_wii.zip)


Is the core "reminiscence_libretro_wii.zip" working? tried here but it occurred to me black screen!

**Comment by Cee123** [comment]
> > Just a couple of other working cores for Wii, not sure why they aren't included.
> > [prosystem_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776150/prosystem_libretro_wii.zip)
> > [reminiscence_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776152/reminiscence_libretro_wii.zip)
> > [smsplus_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776154/smsplus_libretro_wii.zip)
> > [xrick_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776155/xrick_libretro_wii.zip)
> > And of course DosBox is all working now.
> > [dosbox_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776158/dosbox_libretro_wii.zip)
> 
> Is the core "reminiscence_libretro_wii.zip" working? tried here but it occurred to me black screen!

Which file did you load with it? I had to try different game files to load with it. Some worked and some just went to black screen.

**Comment by rogerioronin** [comment]
> > > Just a couple of other working cores for Wii, not sure why they aren't included.
> > > [prosystem_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776150/prosystem_libretro_wii.zip)
> > > [reminiscence_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776152/reminiscence_libretro_wii.zip)
> > > [smsplus_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776154/smsplus_libretro_wii.zip)
> > > [xrick_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776155/xrick_libretro_wii.zip)
> > > And of course DosBox is all working now.
> > > [dosbox_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776158/dosbox_libretro_wii.zip)
> > 
> > 
> > Is the core "reminiscence_libretro_wii.zip" working? tried here but it occurred to me black screen!
> 
> Which file did you load with it? I had to try different game files to load with it. Some worked and others just went to black screen.


It worked here on the Wii, I tested other versions and one of them worked!

**Comment by Cee123** [comment]
This is an issue again. Just tried the latest nightly, and the controls aren't responding with most of the cores except for in the RetroArch menu. Dosbox is gone. Vice freezes on a black screen. And CPS3 crash dumps.

**Comment by Cee123** [comment]
Update: Sorry my apologies. The only cores where the controls don't work seem to be the FB Alpha 2012 cores. All the rest I've tried work fine.

**Comment by Cee123** [comment]
Actually there is quite a bit of bugginess with the cores - things like freezing, crash dumps and other issues.

**Comment by twinaphex** [comment]
Stsrt writing individual Github issue reports per core. That will make this somewhat more manageable.

**Comment by twinaphex** [comment]
Also, reminder again we need maintainers for RetroArch Gamecube/Wii port.

**Comment by Edu789** [comment]
> This is an issue again. Just tried the latest nightly, and the controls aren't responding with most of the cores except for in the RetroArch menu. Dosbox is gone. Mednafen VB is gone. Vice freezes on a black screen. And CPS3 crash dumps.

The Fmsx is also gone and the Frodo hasn't been added yet...

**Comment by Edu789** [comment]
> This is an issue again. Just tried the latest nightly, and the controls aren't responding with most of the cores except for in the RetroArch menu. Dosbox is gone. Mednafen VB is gone. Vice freezes on a black screen. And CPS3 crash dumps.

The Fmsx and Cap32 are also gone and the Frodo hasn't been added yet...

**Comment by Edu789** [comment]
Has anyone tried to compile the O2em and Vecx cores?

**Comment by Cee123** [comment]
This issue was fixed using the input bitmasks with the rest of the cores.. They're all working well now. But the FB Alpha 2012 cores weren't working back then so they didn't get fixed.

**Comment by Cee123** [comment]
> 
> 
> Stsrt writing individual Github issue reports per core. That will make this somewhat more manageable.

You are right, thanks. I might do that with the other bugs I've noticed. Would be a lot easier.

**Comment by Edu789** [comment]
> > > > Just a couple of other working cores for Wii, not sure why they aren't included.
> > > > [prosystem_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776150/prosystem_libretro_wii.zip)
> > > > [reminiscence_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776152/reminiscence_libretro_wii.zip)
> > > > [smsplus_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776154/smsplus_libretro_wii.zip)
> > > > [xrick_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776155/xrick_libretro_wii.zip)
> > > > And of course DosBox is all working now.
> > > > [dosbox_libretro_wii.zip](https://github.com/libretro/RetroArch/files/5776158/dosbox_libretro_wii.zip)
> > > 
> > > 
> > > Is the core "reminiscence_libretro_wii.zip" working? tried here but it occurred to me black screen!
> > 
> > 
> > Which file did you load with it? I had to try different game files to load with it. Some worked and others just went to black screen.
> 
> It worked here on the Wii, I tested other versions and one of them worked!
https://github.com/r-type/frodo-libretro/issues/8#issuecomment-699543256

https://github.com/libretro/blueMSX-libretro/issues/121#issuecomment-695785012

**Comment by Cee123** [comment]
I've opened them a separate issue for each of the cores. But I've linked them back here, so there's a reference of what might need fixing (or not).

**Comment by Cee123** [comment]
I found an old version of Mame2003 Plus compiled for the Wii on GBATemp, and split into several parts (as the main core was too big for the Wii to load). Part C of the Mame2003 Plus Core runs these Midway games with no controller issues whatsoever. Some games produce a black screen. But the ones that do work, work well.

https://www.mediafire.com/file/ptgzojp89t4t26u/MAME2003-WII-0.4.zip/file

**Comment by Cee123** [comment]
Is anyone here? There's only a couple of cores that need the bitmask added and everything should be fully working. It's only a few lines of code as far as I can tell made to the libretro.c file. I'd do it myself if I could figure out how to do it. And I tried to use the others as a reference but unfortunately it didn't work. Hoping that someone with a little bit more knowledge than me could help out.

**Comment by Cee123** [comment]
Man nothing's changed with this since February. :(

**Comment by Cee123** [comment]
I think I see what the problem is with the FBAlpha2012 cores they don't have the files necessary to add those lines of code... On the other hand a bitmask could be added to Mame2003 Midway. And that core itself could be added to the Wii build. I've tested it out, it works well on the Wii... Except for that one problem - the damn controls!

**Comment by LibretroAdmin** [comment]
@Cee123 Is the Wii version still working for you? Where do we stand right now?

**Comment by Cee123** [comment]
> @Cee123 Is the Wii version still working for you? Where do we stand right now?

I will test the latest version to see if the issue is resolved on all the cores. Will do that and mention here if there's any issues or whether everything works well now.

**Comment by Cee123** [comment]
> @Cee123 Is the Wii version still working for you? Where do we stand right now?

Just tested a few things. Things are still the same. Same bugs, same crashes, same cores (I.e. fb alpha and mame cores and probably a few others) where the controls still don't respond. But for most of the cores, they are fixed, but there are still cores where the controls don't work at all.

**Comment by LibretroAdmin** [comment]
OK, list all the issues again and I will try to see what I can do. I assume a lot of this is still related to some cores not implementing button bitmasks?

**Comment by Cee123** [comment]
> OK, list all the issues again and I will try to see what I can do. I assume a lot of this is still related to some cores not implementing button bitmasks?

That's right. I think it's only a few certain cores like the fbalpha and mame cores. But there could be others too. Some cores I've noticed like the Vice cores and now the fbalpha2012 cps3 and neo geo (which used to work before) don't seem to work at all.

**Comment by Cee123** [comment]
> OK, list all the issues again and I will try to see what I can do. I assume a lot of this is still related to some cores not implementing button bitmasks?

Ok from what I've tested 
Fbalpha2012 - controls don't respond 
Fbalpha2012 cps1 - controls don't respond 
Fbalpha2012 cps2 - roms won't load
Fbalpha2012 cps3 - crash dumps
Genesis plus gx - black screen upon loading rom
Same genesis plus gx wide-screen 
Vice cores don't work - freeze on black screen
Fbalpha2012 neo geo - roms won't load (needs a different romset maybe?)
Bitmasks are absent on all Fbalpha2012 cores
Gpsp - loads rom then freezes
Super mario war - unbearable sound issue/hissing/white noise
Fceumm - loads rom then goes to a black screen

All the others I've tried worked fine and had no issues. But there were also various cores I didn't test.

**Comment by LibretroAdmin** [comment]
Is this with the latest nightlies? 

Disappointing to hear that Genesis Plus GX stopped working, especially on the main platform (Wii/Gamecube) it originated from. We should be able to get that fixed.

**Comment by Cee123** [comment]
Just tested this again with the latest nightlies. Controls still not working on a bunch of cores and some of the cores have even stopped working altogether.

**Comment by Cee123** [comment]
Closing - looks like nothing's happening with this

**Comment by LibretroAdmin** [comment]
Do not close issues that have not been resolved.

**Comment by Cee123** [comment]
Since I've pointed this out it's been nearly 3 years now. Someone just needs to add the input bitmasks for the fbalpha and mame cores and then I can close this. Anyone? I can open a separate issue for the other cores that no longer work.

**Comment by Cee123** [comment]
> Is this with the latest nightlies?
> 
> Disappointing to hear that Genesis Plus GX stopped working, especially on the main platform (Wii/Gamecube) it originated from. We should be able to get that fixed.

You said genesis plus gx would be a simple fix since it originated on the Wii. I can confirm that still doesn't work unfortunately.

**Comment by LibretroAdmin** [comment]
We have no development kits right now for Wii. We would need more Wii contributors. As it is unfortunately, most in that scene are only end-users.

We can implement bitmasks for cores that are still missing it although that does beg the question why it won't just work without bitmasks though, it should. Something is perhaps wrong there on the Wii input driver end.

**Comment by Cee123** [comment]
Ah ok that makes sense. That's no problem then, I understand. Yeah it is interesting because it never needed those implemented before, so you could be right about the drivers. I noticed the problem going back to when the buildbot was hacked. Very strange.

**Comment by Cee123** [comment]
I just checked the finalburn alpha cores for the wii again today and can confirm the controls still don't work for these cores - four years later. Three of the cores crash - the cps2, cps3 and neo geo cores. Are these broken or am I doing something wrong?

**Comment by YoPaeisan** [comment]
Hi all, I first just want to applaud the efforts of all the devs! As someone who is mostly an end-user thank you so much! 

I was struggling with this polling issue over the weekend, what made it all the more frustrating is that I had the Wii and nunchuk emulating the CPS2 D&D ROM/input perfectly in the past, successfully mapping all of the wiimote/nunchuk buttons (specifically z,c, +, and - not polling when remapping).

I'd like to contribute more, however I am not savvy enough to branch away from the current version, however I was able to find a Wii port of Retroarch 1.7.5 that I had backed up, and sure enough still works. 

I did some experimenting with the new CPS2 core and tried using the old .CFG to get the inputs working with the newest RA, but no progress worth mentioning. 

I'd love to help work on a patch, but I haven't seen the repository for 1.7.5 so I'm not sure how to peak at the working code. 

Anyway, for anyone struggling with this issue glhf I hope this helps!

**Comment by Cee123** [comment]
That's the problem. The controls have been pretty much useless for these since that particular version and that was years ago. No one even looked at it at the time, and no one has cared about it since. I've since lost interest in this and gave up hoping for a solution. I have old versions of FBAlpha2012 cores that I use before the controller issue, since the versions after that haven't been fixed or even looked at, ever. I did manage to find old versions of the core on the internet archive and those work too. I was hoping that someone would look at this issue when I first noticed and reported it because it would've been easier to compare between working and non-working versions back then to see what was going on. Now that's more difficult because you have to search for an old version that works and the code for it.

**Comment by Cee123** [comment]
I got ChatGPT to write a script for the controller mapping but I didn't know how to incorporate it into the emulator or even how to test it.

**Comment by YoPaeisan** [comment]
I didn't think to use chatGPT, that's clever! I was going to see if I could use a wayback machine to access the 1.7.5 repository and compare with the current input scripts.

**Comment by YoPaeisan** [comment]
As far as implementing your code: I'm a noob when it comes to open source dev and github, but in general you need to set up a workflow so you can branch off whatever build and compile/test your edits. 

I have limited experience but my interest/background is in assembly. There a couple guys out there publishing how-to's for compiling nes roms. I started there. I would expect much of the same nuance at the core of the RA engine.

We know RA is reading the nunchuk input, it's somewhere between polling the input and passing it along to the script for remapping.

**Comment by Cee123** [comment]
There was a guy here who fixed the controller issue for the majority of cores and he did a fantastic job. But unfortunately one of the guys in charge was really rude to him, saying that it wasn't needed and that there was no issues with the controls etc and he left. I've since given up on this. This will likely stay broken forever and I don't care anymore. It's their project and if that's how they want it to stay, then that's fine. I reported it ages ago. Now it's hard to find the code for the working version of the core since its been so long now.

**Comment by Cee123** [comment]
> I didn't think to use chatGPT, that's clever! I was going to see if I could use a wayback machine to access the 1.7.5 repository and compare with the current input scripts.

The guy who fixed the majority of these cores implemented something called an input or controller bitmask. He did it for each one of these cores. Unfortunately I think most libretro cores need it now (for some unknown reason) except for the ones on this list that he's fixed. After he left, I tried to do it myself for some other cores but I just couldn't do it or get any working. Even ChatGPT wasn't giving me a clear and concise answer on how to do it.

**Comment by YoPaeisan** [comment]
No luck finding the older repository, but I can see where the devs are coming from, the Wii hardware can only emulate so much so why invest any time on solving problems with that port of RA? I actually found RA 1.7.6 posted with the split MAME 0.78 cores (mame2003+). The core is split to accommodate the Wii's memory limitations and apparently that workaround made a bunch more ROMs playable. It was a pretty deep rabbit hole but I got a few games working. 

I'm curious to know more about how the RA core is ported to the various systems it can run on. I would start there and try to find where the data input from the wiimote  is being stored as input for the emulator cores. 

What that other guy was doing with the emulator cores is interesting, but I don't think that's the best solution. It could have something to do with a bit mask though. 

I remember something about bitmasks doing nesdev: essentially every bit of a byte represents a different button on the nes remote so your program checks each bit by masking the others out to determine what's being pressed before the next frame is rendered. 

We know RA is getting the Wiimote button presses, but it's not passing them all into the remap function.

**Comment by Cee123** [comment]
It did have everything to do with the bitmasks. That's why the controls are even working on a lot of the cores now. They weren't working before. It happened right after the time that libretro's buildbot got hacked, thats when i noticed this issue with every core suddenly happening. Almost like something got corrupted when that happened. He fixed the majority of those listed by implementing those bitmasks, while one guy was telling him not to and that it was pointless. It wasnt pointless. That was the only thing that got the controllers working for those. That other guy never provided an alternative solution other than criticism. A couple of the cores still crash from that time as well. The neo geo core no longer works and it used to as well as the CPS3 core, so I've had to use old versions of those too before the buildbot hack. I've reported those on the github too. But yeah it's very, very strange. I tried looking at the code but couldn't see what the problem could be.

**Comment by YoPaeisan** [comment]
The thing with coding, there's usually more than one way to solve a problem. I get that patching the cores was having an effect, but applying the same patch to every core seems heavy handed. Admittedly my familiarity with the code here is limited. I learned that when you're using the mame cores you have to have romsets with versions that match the version of mame you're using. Maybe that's why your games started crashing? It's too bad the only guy getting results was discouraged from continuing. Feedback is valuable, but criticism that isn't constructive is just pointless.

**Comment by YoPaeisan** [comment]
I dropped out of comp. ed. in grade 12, so I'm starting at a grade eleven level of computer education (which means, for me, no c/c++ experience.) But from what I gleaned this morning I think getting the Wiimote/nunchuk working on (my) PC would be a faster route.

I ran into a similar problem a few years back: I was emulating nes on a smart phone, using my Wiimote as my input, until (after an Android update) I could no longer poll the Wiimote with my phone's radio. I can't remember the specifics of what changed, but the issue was in a change to the protocol used to connect to the Wiimote. Maybe it has something to do with the way the current API translates bluetooth controller data for the emulator?

Not sure if it's a problem to connect Wiimote to PC or not? I must find out.
